tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -2034,6 +2034,26 @@ void HWCommandQueue::enqueue_program(Program& program, bool blocking) {
         expected_workers_completed);
     this->enqueue_command(command, blocking);
 
+    if (program.has_multi_device_dependencies() and not this->device->is_mmio_capable() and tt::Cluster::instance().is_galaxy_cluster() and not this->tid.has_value()) {
```

Comment:
Is this not exposable on T3k with 2 cqs? Also what's this check on tid?

Reply:
This particular deadlock is not exposable on T3K. It only occurs when running CCLs on multiple remote devices in a tunnel.
A similar deadlock is seen on T3K/N300 with 2 CQs and event synchronization. Tests and functionality to resolve those deadlocks are already on main